src/Endpoints/class-referrers-post-detail-api-proxy.php

```diff
@@ -0,0 +1,217 @@
+		$result->totals->views = $total_views;
+
+		// Set percentage values and format numbers.
+		foreach ( $result as $key => $value ) {
+			// Set and format percentage values.
+			$result->{ $key }->viewsPercentage = $this->get_i18n_percentage(
+				(int) $value->views,
+				(int) $total_views
+			);
+
+			// Format views values.
+			$result->{ $key }->views = number_format_i18n( $result->{ $key }->views );
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Generates the top referrers data.
+	 *
+	 * Returned object properties:
+	 * - `views`:          The number of views.
+	 * - `viewPercentage`: The number of views as a percentage, compared to the
+	 *                     total views of all referrer types.
+	 * - `datasetViewsPercentage: The number of views as a percentage, compared
+	 *                            to the total views of the current dataset.
+	 *
+	 * @since 3.6.0
+	 *
+	 * @param int             $limit The limit of returned referrers.
+	 * @param array<stdClass> $response The response received by the proxy.
+	 * @param int             $total_views The total views of the $views object.
+	 * @return stdClass The generated data.
+	 */
+	private function generate_referrers_data(
+		int $limit, array $response, int $total_views
+	): stdClass {
+		$result = new stdClass();
+		$totals = 0;
+
+		// Set the new limit if too few results are returned.
+		$response_item_count = count( $response );
+		if ( $response_item_count < $limit ) {
+			$limit = $response_item_count;
+		};
+
+		// Set views and views totals.
+		for ( $i = 0; $i < $limit; $i++ ) {
+			$data  = $response[ $i ];
+			$views = $data->metrics->referrers_views;
+
+			$result->{ $data->name }->views = $views;
+			$totals                        += $views;
+		}
+
+		// Add totals to the end of the object.
+		$result->totals->views = $totals;
+
+		// Set percentages values and format numbers.
+		foreach ( $result as $key => $value ) {
+			// Percentage against all referrer views, even those not included
+			// in the dataset due to the $limit argument.
+			$result->{ $key }->viewsPercentage = $this
+				->get_i18n_percentage( (int) $value->views, $total_views );
+
+			// Percentage against the current dataset that is limited due to the
+			// $limit argument.
+			$result->{ $key }->datasetViewsPercentage = $this
+				->get_i18n_percentage( (int) $value->views, $totals );
+
+			// Format views values.
+			$result->{ $key }->views = number_format_i18n( $result->{ $key }->views );
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Returns the passed number compared to the passed total, in an
+	 * internationalized percentage format.
+	 *
+	 * @since 3.6.0
+	 *
+	 * @param int $number The number to be calculated as a percentage.
+	 * @param int $total The total number to compare against.
+	 * @return string The resulting internationalized percentage.
+	 */
+	private function get_i18n_percentage( int $number, int $total ) {
+		return number_format_i18n( $number / $total * 100, 2 );
```

Comment:
If the `$total` is 0, then this will generate a Division by Zero fatal. Add a guard clause for that condition.

Reply:
This opens-up the broader question of what we would need to do even in the UI side in case of zero views/totals. I'll need to dig a bit more into this.

Reply:
I've added a guard against this in that function, and noted about the UI so I can look into it later. I'm marking this as resolved, but feel free to comment if needed.